### PR TITLE
k8s: error if we can't watch _any_ resources

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: internal/k8s/watch.go [hunk 1]
```diff
@@ -294,6 +294,10 @@ func (kCli K8sClient) watchableGroupVersionResources(ctx context.Context) ([]sch
 		}
 	}
 
+	if len(ret) == 0 {
```
Comment (maiamcc) [inline internal/k8s/watch.go]: didn't we say that it made more sense to check the number of things returned in the caller of this func? though meh i don't have strong feels

File: internal/k8s/watch_test.go [hunk 2]
```diff
@@ -259,6 +264,11 @@ func (tf *watchTestFixture) runServices(input []runtime.Object, expectedOutput [
 	assert.Equal(tf.t, expectedOutput, observedServices)
 }
 
+func (tf *watchTestFixture) watchEverythingExpectError(expectedErr string) {
+	_, err := tf.kCli.WatchEverything(tf.ctx, []model.LabelPair{})
```
Comment (maiamcc) [inline internal/k8s/watch_test.go]: I'm not sure I get why this errors -- can you add a comment about why?

File: internal/k8s/watch.go (hunk 1 above)
Comment (jazzdan) [reply internal/k8s/watch.go]: Yeah, the problem is the thing that calls this returns a channel, and you can't exactly check the length of a channel. I could move it up one level though.

File: internal/k8s/watch_test.go (hunk 2 above)
Comment (jazzdan) [reply internal/k8s/watch_test.go]: Sure!

**Pull request merged**